MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs

```diff
@@ -933,6 +933,18 @@ private void PlatformSetViewport(ref Viewport value)
         }
 
         // Only implemented for DirectX right now, so not in GraphicsDevice.cs
+        public void SetRenderTarget(RenderTarget2D renderTarget, int arraySlice)
+        {
+            if (!GraphicsCapabilities.SupportsTextureArrays)
+                throw new InvalidOperationException("Texture arrays are not supported on this graphics device");
+
+            if (renderTarget == null)
+                SetRenderTarget(null);
+            else
+                SetRenderTargets(new RenderTargetBinding(renderTarget, arraySlice));
+        }
+
+        // Only implemented for DirectX right now, so not in GraphicsDevice.cs
         public void SetRenderTarget(RenderTarget3D renderTarget, int arraySlice)
```

Comment:
Yep, in HLSL, there's certainly a TextureCube type. But on the D3D side, there's no TextureCube type - only a Texture2D with arraySize == 6 and BindFlags == BindFlags.TextureCube.

That said, I don't think it's a problem for MG to have both. The fact that one is a special case of the other is a D3D (and maybe actual hardware?) implementation detail. It's not the same in OpenGL. I don't know about Vulkan / D3D12, but we can cross that bridge later.

For now, I think having TextureCube / RenderTargetCube classes matches what users expect.